keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -239,21 +239,35 @@ open class SqlResourceRepository(
     jooq.transaction { config ->
       val txn = DSL.using(config)
 
+      val resourceUid = txn.select(RESOURCE.UID).from(RESOURCE).where(RESOURCE.ID.eq(event.id)).fetchOne(RESOURCE.UID)
+
       if (event.ignoreRepeatedInHistory) {
         val previousEvent = txn
           .select(EVENT.JSON)
-          .from(EVENT, RESOURCE)
+          .from(EVENT)
           .where(EVENT.SCOPE.eq(Scope.RESOURCE.name))
-          .and(RESOURCE.ID.eq(event.id))
-          .and(RESOURCE.UID.eq(EVENT.UID))
+          .and(EVENT.UID.eq(resourceUid))
           .orderBy(EVENT.TIMESTAMP.desc())
           .limit(1)
           .fetchOne()
           ?.let { (json) ->
             objectMapper.readValue<ResourceEvent>(json)
           }
 
-        if (event.javaClass == previousEvent?.javaClass) return@transaction
+        // If an equivalent event is already the latest in history, just update the timestamp
+        if (event.javaClass == previousEvent?.javaClass) {
+          txn
+            .update(EVENT)
+            .set(EVENT.TIMESTAMP, event.timestamp.atZone(clock.zone).toLocalDateTime())
+            // we udpate the JSON as well because it also contains a timestamp
```

Comment:
Oof, you're right. That is a recipe for bugs. Will you create an issue and we can rethink it (not for this pr obviously)?